tests/BuildScriptGenerator.Tests/DefaultBuildScriptGeneratorTest.cs

```diff
@@ -617,7 +620,9 @@ public void Checkers_DontFailTheBuild_WhenTheyThrow()
             commonOptions = commonOptions ?? new BuildScriptGeneratorOptions();
             commonOptions.SourceDir = "/app";
             commonOptions.DestinationDir = "/output";
-
+            var telemetryClientMock = new Mock<TelemetryClientMock>();
```

Comment:
Is there a reason we can't just do `Mock<TelemetryClient>` here?

Reply:
Syced with @Cormac offline. Since, `TelemetryClient `is a sealed class, I created a static helper class `TelemetryClientHelper` that would return a telemetry client object when the `GetTelemetryClient` method is called.